galaxy/management/commands/generate_galaxy_test_data.py

```diff
@@ -0,0 +1,288 @@
+                unix_name=f"galaxy-club-{i}", name=f"club-{i}"
+            )
+
+    def make_users(self):
+        # This will create all the users and store them in self.users for fast access later
+        self.users = {}
+        for i in range(self.NB_USERS):
+            u = User.objects.create_user(
+                username=f"galaxy-user-{i}",
+                email=f"{i}@galaxy.test",
+                first_name="Citizen",
+                last_name=f"n°{i}",
+            )
+            self.users[i] = u
+
+            self.logger.info(f"Registering {u}")
+            Subscription.objects.create(
+                member=u,
+                subscription_start=Subscription.compute_start(
+                    self.now - timedelta(days=self.NB_USERS - i)
+                ),
+                subscription_end=Subscription.compute_end(duration=2),
+            )
+
+    def make_families(self):
+        # This will iterate on all citizen after the 200th.
+        # Then it will take 14 other citizen among the 200 preceding (godfathers are usually older), and apply another heuristic to determine if they should have a family link
+        for i in range(200, self.NB_USERS):
+            for j in range(i - 200, i, 14):  # this will loop 14 times (14² = 196)
+                if (i / 10) % 10 == (i + j) % 10:
+                    u1 = self.users[i]
+                    u2 = self.users[j]
+                    self.logger.info(f"Making {u2} the godfather of {u1}")
+                    u1.godfathers.add(u2)
+
+    def make_club_memberships(self):
+        # This function makes multiple passes on all users to affect them some pseudo-random roles in some clubs.
+        # The multiple passes are useful to get some variations over who goes where.
+        # Each pass for each user has a chance to affect her to two different clubs, increasing a bit more the created chaos, while remaining purely deterministic.
+        for i in range(1, 11):  # users can be in up to 20 clubs
+            self.logger.info(f"Club membership, pass {i}")
+            for uid in range(
+                i, self.NB_USERS, i
+            ):  # Pass #1 will make sure every user is at least in one club
+                user = self.users[uid]
+                club = self.clubs[(uid + i**2) % self.NB_CLUBS]
+
+                start = self.now - timedelta(
+                    days=(((self.NB_USERS - uid) * i) // 110)
+                )  # older users were in clubs before newer users
+                end = start + timedelta(days=180)  # about one semester
+                self.logger.info(
+                    f"Making {user} a member of club {club} from {start} to {end}"
+                )
+                Membership(
+                    user=user,
+                    club=club,
+                    role=(uid + i) % 10 + 1,  # spread the different roles
+                    start_date=start,
+                    end_date=end,
+                ).save()
```

Comment:
Same remark : `bulk_create` instead of `save`

Reply:
Changed